File: ProdTests/demos/demo_ccm/java/src/CCMClientImpl.java
/* =========================================================================== *
 * This file is part of CARDAMOM (R) which is jointly developed by THALES
 * and SELEX-SI.
 * 
 * It is derivative work based on PERCO Copyright (C) THALES 2000-2003.
 * All rights reserved.
 * 
 * CARDAMOM is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Library General Public License as published by the
 * Free Software Foundation; either version 2 of the License, or (at your
 * option) any later version.
 * 
 * CARDAMOM is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Library General Public
 * License for more details.
 * 
 * You should have received a copy of the GNU Library General
 * Public License along with CARDAMOM; see the file COPYING. If not, write to
 * the Free Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 * =========================================================================== */


package cdmw.basicdemo;

import cdmw.common.Assert;
import cdmw.testutils.Testable;

//
// IDL:thalesgroup.com/BasicDemo/CCM_Client:1.0
//
public class CCMClientImpl extends org.omg.CORBA.LocalObject
    implements com.thalesgroup.UserBasicDemo.User_CCM_Client {

    /**
     * This class provides an object to test the ccm_activate.
     *
     */
    private class CCMClientActivateTest extends Testable {

        private CCMClientImpl client;
        
        public CCMClientActivateTest(CCMClientImpl client) {
            this.name = "Test ccm_activate on Client";
            this.out = System.out;
            this.client = client;
        }
    
        public void doTests() {
            Assert.check(this.client != null);
            setNbOfRequestedTestOK(5);
            
            com.acme.BasicDemo.Registration registerServer = null;
            com.acme.BasicDemo.CCM_Client_Context clientContext = 
                this.client.getContext();

            println("Trying get_connection_to_register...");
            boolean testOk = false;
            try {
                registerServer = clientContext.get_connection_to_register();

                if (registerServer != null) {
                    testOk = true;
                }
            } catch (Exception e) {
                println("Unexpected exception raised!");
            }
            check(testOk);
    
            println("Registering client...");
            testOk = false;
            try {
                if (registerServer != null) {
                    String clientName = registerServer.register();
                    this.client.setName(clientName);
                    testOk = true;
                }
            } catch (Exception e) {
                println("Unexpected exception raised!");
            }
            check(testOk);

            println("Connecting to the display facet...");
            com.acme.BasicDemo.Display display = null;
            testOk = false;
            try {
                display = clientContext.get_connection_to_server();
                if (display != null) {
                    testOk = true;
                }
            } catch (Exception e) {
                println("Unexpected exception raised!");
            }
            check(testOk);

            println("Printing 'hello' via the server...");
            testOk = false;
            try {
                if (registerServer != null) {
                    display.print(this.client.getText());
                    testOk = true;
                }
            } catch (Exception e) {
                println("Unexpected exception raised!");
            }
            check(testOk);

            println("Printing ValueType hello via the server...");
            testOk = false;
            try {
                if (registerServer != null) {
                    com.acme.BasicDemo.Event event = new EventImpl();
                    event.text = "Hello ValueType!";
         
                    display.print_event(event);
                    testOk = true;
                }
            } catch (Exception e) {
                println("Unexpected exception raised!");
            }
            check(testOk);
        }

    }

    private String name;
    
    private String text;
    
    private com.acme.BasicDemo.CCM_Client_Context sessionContext;

    public CCMClientImpl() {
        this.name = "";
        this.sessionContext = null;
    }

    //
    // IDL:thalesgroup.com/BasicDemo/CCM_Client_Executor/name:1.0
    //
    public String name() {
        return this.name;
    }
     
    //
    // IDL:thalesgroup.com/BasicDemo/CCM_Client_Executor/event:1.0
    //
    //public com.acme.BasicDemo.Event event() {
    //    com.acme.BasicDemo.Event event = new EventImpl();
    public com.acme.BasicDemo.Event2 event() {
        com.acme.BasicDemo.Event2 event = new com.acme.BasicDemo.Event2();
        event.text = this.text;
        return event;
    }

    public void event(com.acme.BasicDemo.Event2 event) {
        this.text = event.text;
    }

    public void set_session_context(org.omg.Components.SessionContext ctx) {
        System.out.println("setSessionContext called!");
        this.sessionContext = 
            com.acme.BasicDemo.CCM_Client_ContextHelper.narrow(ctx);
    }

    public void ccm_activate() {
        CCMClientActivateTest test = new CCMClientActivateTest(this);
        test.start();
    }
    
    public void ccm_passivate() {
        System.out.println("ccm_passivate called!");
    }
    
    public void ccm_remove() {
        System.out.println("ccm_remove called!");
    }

     // additional operation
     
    public void setName(String name) {
        this.name = name;
    }
    
    public com.acme.BasicDemo.CCM_Client_Context getContext() {
        return this.sessionContext;
    }

    public String getText() {
        return this.text;
    }

}

File: tags/CARDAMOM_V1_1_CE/CCMContainer/ccmcif/test/testmonoccmcif/c++/src/CCM_Client_impl.cpp
/* ===================================================================== */
/*
 * This file is part of CARDAMOM (R) which is jointly developed by THALES 
 * and SELEX-SI. 
 * 
 * It is derivative work based on PERCO Copyright (C) THALES 2000-2003. 
 * All rights reserved.
 * 
 * CARDAMOM is free software; you can redistribute it and/or modify it under 
 * the terms of the GNU Library General Public License as published by the
 * Free Software Foundation; either version 2 of the License, or (at your 
 * option) any later version. 
 * 
 * CARDAMOM is distributed in the hope that it will be useful, but WITHOUT 
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or 
 * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Library General Public 
 * License for more details. 
 * 
 * You should have received a copy of the GNU Library General 
 * Public License along with CARDAMOM; see the file COPYING. If not, write to 
 * the Free Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
*/
/* ===================================================================== */

#include "testmonoccmcif/CCM_Client_impl.hpp"
#include "testmonoccmcif/Event_impl.hpp"

#include "Foundation/testutils/Testable.hpp"
#include "Foundation/common/Assert.hpp"


namespace 
{

/**
*  This class provides an object to test the ccm_activate.
*
**/

class test_ccm_client_activate : public Cdmw::TestUtils::Testable
{
  BasicDemo::CCM_Client_impl* m_client;
  
 public:
  test_ccm_client_activate(BasicDemo::CCM_Client_impl* client)
      : Testable("Test ccm_activate on Client"),
        m_client(client)
  {
  };

  ~test_ccm_client_activate() {};

  void do_tests()
  {
     // set number of requested successfull tests
     set_nbOfRequestedTestOK (5);
    
     CDMW_ASSERT(m_client);

      ::BasicDemo::Registration_var register_server;
     ::BasicDemo::CCM_Client_Context_var client_context = m_client->get_context();
      
     TEST_INFO("Get_connection_to_register...");
      
      bool test_ok = false;
      try
      {
          register_server = client_context->get_connection_to_register();

            if (!CORBA::is_nil(register_server.in()))
            {
                test_ok = true;
            }
      }
      catch (...)
      {
          TEST_INFO("Unexpected exception raised!");
      }
      TEST_CHECK(test_ok);

      // register client
      TEST_INFO("Register client...");
     test_ok = false;
      try
      {
             if (!CORBA::is_nil(register_server.in()))
            {
              CORBA::String_var client_name = register_server->_cxx_register();
              m_client->set_name(client_name.in());
                test_ok = true;
         }
      }
      catch (...)
      {
            TEST_INFO("Unexpected exception raised!");
      }
      TEST_CHECK(test_ok);
     
     //get the connection to the server 
      TEST_INFO("Connect to the display facet...");
      ::BasicDemo::Display_var display;
     test_ok = false;
      try
      {
         display = client_context->get_connection_to_server();

            if (!CORBA::is_nil(display.in()))
            {
                test_ok = true;
            }
      }
      catch (...)
      {
            TEST_INFO("Unexpected exception raised!");
      }
      TEST_CHECK(test_ok);

     //print "hello" via the server
      TEST_INFO("Print hello...");
     test_ok = false;
      try
      {
             if (!CORBA::is_nil(register_server.in()))
            {
            display->print(m_client->get_text());
                test_ok = true;
            }
      }
      catch (...)
      {
            TEST_INFO("Unexpected exception raised!");
      }
      TEST_CHECK(test_ok);

     //print_event "hello" via the server
      TEST_INFO("Print ValueType hello...");
     test_ok = false;
      try
      {
             if (!CORBA::is_nil(register_server.in()))
            {
             BasicDemo::Event_var event = new BasicDemo::Event_impl();
             event->text("Hello ValueType!");
             
             display->print_event(event.in());
             test_ok = true;
            }
      }
      catch (...)
      {
            TEST_INFO("Unexpected exception raised!");
      }
      TEST_CHECK(test_ok);
  };
}; // End of class test_ccm_client_activate
    
} // End of anonymous namespace



//
// IDL:thalesgroup.com/BasicDemo/CCM_Client:1.0
//
BasicDemo::CCM_Client_impl::CCM_Client_impl()
    : m_name(""),
      m_session_context(BasicDemo::CCM_Client_Context::_nil())
{
}

BasicDemo::CCM_Client_impl::~CCM_Client_impl()
{
    std::cout << "CCM_Client_impl::~CCM_Client_impl called!" << std::endl;
}


//
// set_session_context function : it fixes the session context of the component (call by the container)
//
void
BasicDemo::CCM_Client_impl::set_session_context(Components::SessionContext_ptr ctx)
    throw(CORBA::SystemException)
{
    std::cout << "Set_session_context called!" << std::endl;
   
    m_session_context = ::BasicDemo::CCM_Client_Context::_narrow(ctx); 
}


//
// ccm_activate function : it actives the component (call by the container)
//
void
BasicDemo::CCM_Client_impl::ccm_activate()
   throw(CORBA::SystemException)
{

    test_ccm_client_activate test(this);

    test.start();
}


//
// ccm_passivate function : it makes stop the component (call by the container)
//

void
BasicDemo::CCM_Client_impl::ccm_passivate()
   throw(CORBA::SystemException)
{
   std::cout << "Ccm_passivate called!" << std::endl;
} 



//
// ccm_remove function : it destroys the component (call by the container)
void
BasicDemo::CCM_Client_impl::ccm_remove()
   throw(CORBA::SystemException)
{
   std::cout << "Ccm_remove called!" << std::endl;
}


//
// IDL:thalesgroup.com/BasicDemo/CCM_Client_Executor/name:1.0
//
char*
BasicDemo::CCM_Client_impl::name()
    throw(CORBA::SystemException)
{
    char* name = CORBA::string_dup(m_name.in());
    return name;
}

//
// IDL:thalesgroup.com/BasicDemo/CCM_Client_Executor/event:1.0
//
BasicDemo::Event* 
BasicDemo::CCM_Client_impl::event()
    throw(CORBA::SystemException)
{
    BasicDemo::Event* event = new BasicDemo::Event_impl();
    event->text(m_text.in());

    return event;
}

void 
BasicDemo::CCM_Client_impl::event(BasicDemo::Event* event)
    throw(CORBA::SystemException)
{
    m_text = CORBA::string_dup(event->text());
}

